integration/update_field_compat_test.go

```diff
@@ -666,9 +666,24 @@ func TestUpdateFieldCompatSet(t *testing.T) {
 			update:        bson.D{{"$set", bson.D{{"foo", int32(12)}, {"v", nil}}}},
 			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/1676",
 		},
-		"SetSameValueInt": {
-			update: bson.D{{"$set", bson.D{{"v", int32(42)}}}},
-			skip:   "https://github.com/FerretDB/FerretDB/issues/1662",
+		"SameValueInt32Type": {
+			update:        bson.D{{"$set", bson.D{{"v", int32(42)}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/1668",
+		},
+		"SameValueInt64Type": {
+			update:        bson.D{{"$set", bson.D{{"v", int64(42)}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/1668",
+		},
+		"SameValueDoubleType": {
+			update:        bson.D{{"$set", bson.D{{"v", 42.0}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/1668",
+		},
+		"DocSameNumberType": {
+			update: bson.D{{"$set", bson.D{{"v", bson.D{{"foo", int32(42)}}}}}},
+		},
+		"DocDifferentNumberType": {
+			update:        bson.D{{"$set", bson.D{{"v", bson.D{{"foo", int64(42)}}}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/1668",
```

Comment:
This one doesn't need an issue number but an explanation that it can't be set for Tigris. If we attempt to set a different type, Tigris will return an error.

Reply:
:+1: